crates/devices/virtio-blk/src/request.rs

```diff
@@ -297,34 +297,26 @@ mod tests {
         descs: &[Descriptor],
     ) -> DescriptorChain<&'a GuestMemoryMmap> {
         // Support a max of 16 descriptors for now.
-        let vq = VirtQueue::new(GuestAddress(0), mem, 16);
+        let vq = MockSplitQueue::new(GuestAddress(0), mem, 16);
         for (idx, desc) in descs.iter().enumerate() {
             let i = idx as u16;
-            vq.dtable(i).addr().store(desc.addr().0);
-            vq.dtable(i).len().store(desc.len());
-
-            if idx == descs.len() - 1 {
-                // Clear the NEXT flag if it was set. The value of the next field of the
-                // Descriptor doesn't matter at this point.
-                vq.dtable(i)
-                    .flags()
-                    .store(desc.flags() & !VIRTQ_DESC_F_NEXT);
+            let addr = desc.addr().0;
+            let len = desc.len();
+            let (flags, next) = if idx == descs.len() - 1 {
+                (desc.flags() & !VIRTQ_DESC_F_NEXT, 0)
             } else {
-                // Ensure the next flag is set.
-                vq.dtable(i).flags().store(desc.flags() | VIRTQ_DESC_F_NEXT);
-                // Ensure we are referring the following descriptor. This ignores
-                // any value is actually present in `desc.next`.
-                vq.dtable(i).next().store(i + 1);
-            }
+                (desc.flags() | VIRTQ_DESC_F_NEXT, i + 1)
+            };
+
+            let desc = Descriptor::new(addr, len, flags, next);
+            vq.desc_table.store(i, desc);
```

Comment:
Hmm, wondering whether we should use accessors (i.e.` vq.desc_table()`) instead of pub members, and also whether things are more straight forward if we add more helpers (i.e. something like the indexing operator for `DescriptorTable` for easier access to individual descriptors). Let's have a quick chat about this.